.github/workflows/wheels-build.yml

```diff
@@ -0,0 +1,247 @@
+    inputs:
+      default-action:
+        description: >-
+          The default action for each artifact.
+          Choose from 'build' (default) or 'skip'.
+        type: string
+        default: "build"
+        required: false
+
+      sdist:
+        description: >-
+          The action to take for the sdist.
+          Choose from 'default', 'build' or 'skip'.
+        type: string
+        default: "default"
+        required: false
+
+      wheels-tier-1:
+        description: >-
+          The action to take for Tier 1 wheels.
+          Choose from 'default', 'build' or 'skip'.
+          This builds multiple artifacts, which all match 'wheels-tier-1-*'.
+        type: string
+        default: "default"
+        required: false
+
+      wheels-32bit: 
+        description: >-
+          The action to take for Tier 1 wheels.
+          Choose from 'default', 'build' or 'skip'.
+          This builds multiple artifacts, which all match 'wheels-32bit-*'.
+        type: string
+        default: "default"
+        required: false
+
+      wheels-linux-s390x: 
+        description: >-
+          The action to take for Linux s390x wheels.
+          Choose from 'default', 'build' or 'skip'.
+        type: string
+        default: "default"
+        required: false
+
+      wheels-linux-ppc64le: 
+        description: >-
+          The action to take for Linux ppc64le wheels.
+          Choose from 'default', 'build' or 'skip'.
+        type: string
+        default: "default"
+        required: false
+
+      wheels-linux-aarch64: 
+        description: >-
+          The action to take for Linux AArch64 wheels.
+          Choose from 'default', 'build' or 'skip'.
+        type: string
+        default: "default"
+        required: false
+
+      artifact-prefix:
+        description: "A prefix to give all artifacts uploaded with 'actions/upload-artifact'."
+        type: string
+        default: ""
+        required: false
+
+      python-version:
+        description: "The Python version to use to host the build runner."
+        type: string
+        default: "3.10"
+        required: false
+
+      pgo:
+        description: "Whether to enable profile-guided optimizations for supported platforms."
+        type: boolean
+        default: true
+        required: false
+      
+
+jobs:
+  wheels-tier-1:
+    name: "Wheels / Tier 1"
+    if: (inputs.wheels-tier-1 == 'default' && inputs.default-action || inputs.wheels-tier-1) == 'build'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          # Used for the x86_64 builds.
+          - macos-12
+          # Used for the ARM builds.
+          - macos-14
+          - windows-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ inputs.python-version }}
+          architecture: ${{ matrix.os == 'macos-14' && 'arm64' || 'x64' }}
```

Comment:
Also here, this is how I interpret it:
```txt
If matrix.os current value is "macos-14"  and architecture is  either "arm64" or "x64".
```
Is the second clause supposed to give you the only two possible options?

Reply:
This is equivalent to the Python:
```python
architecture = "arm64" if matrix.os == "macos-14" else "x64"
```
(since here there's no worries about the empty string appearing - both are string literals).

The GitHub Actions expression DSL gives us unfortunately few ways to do a _proper_ conditional within a single expression.